### ISSUE TITLE
Create tooling for visualising the dependency graph, inner workings of dependency algorithm and graph processing

### ISSUE THREAD

**Issue by safesparrow** [safesparrow/fsharp#16]
It would be really nice to have an easy way to draw graphs showing:
1. file dependency graph (most important)
2. inner workings of the dependency resolution algorithm (eg. the initial Trie)
3. inner workings of the graph processing algorithm (eg. showing how nodes are being processed in parallel)

A nice way to do it would be to:
1. have code dump the graph/a list of steps in a textual format
2. a simple Python script/notebook that produces [interactive] graphs
